syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.github.yzernik.squeakserver";
option java_outer_classname = "SqueakAdminProto";
option objc_class_prefix = "SQK";

package squeakserver;

// import "lnd.proto";

// Interface exported by the server.
service SqueakAdmin {

  /** sqkadmin: `getsqueak`
  */
  rpc GetBalance (GetBalanceRequest) returns (GetBalanceReply) {}

  /** sqkadmin: `createsigningprofile`
  */
  rpc CreateSigningProfile (CreateSigningProfileRequest) returns (CreateSigningProfileReply) {}

  /** sqkadmin: `getsqueakprofile`
  */
  rpc GetSqueakProfile (GetSqueakProfileRequest) returns (GetSqueakProfileReply) {}

  /** sqkadmin: `makesqueak`
  */
  rpc MakeSqueak (MakeSqueakRequest) returns (MakeSqueakReply) {}

}

message GetBalanceRequest {
}

message GetBalanceReply {
    /// The wallet balance
    int64 balance = 1;
}

message CreateSigningProfileRequest {
    /// The name of the new signing profile
    string profile_name = 1;
}

message CreateSigningProfileReply {
    /// The profile id
    int32 profile_id = 1;
}

message GetSqueakProfileRequest {
    /// The profile id
    int32 profile_id = 1;
}

message GetSqueakProfileReply {
    /// The squeak profile
    SqueakProfile squeak_profile = 1;
}

message SqueakProfile {
    /// The profile id
    int32 profile_id = 1;

    /// The profile name
    string profile_name = 2;

    /// The private key
    bytes private_key = 3;

    /// The address
    string address = 4;

    /// Sharing
    bool sharing = 5;

    /// Following
    bool following = 6;
}

message MakeSqueakRequest {
    /// The profile id
    int32 profile_id = 1;

    /// The content
    string content = 2;

    /// The replyto hash
    bytes replyto = 3;
}

message MakeSqueakReply {
    /// Hash of the created squeak.
    bytes hash = 1;
}
